### PR TITLE
feat: Initialize EVM app structure

### PR DESCRIPTION
**Pull request by natto1784**
https://app.clickup.com/t/9002019994/PRF-4894
https://app.clickup.com/t/9002019994/PRF-4895

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4894 EVM application skeleton](https://app.clickup.com/t/9002019994/PRF-4894)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4895 Data flow handling APIs](https://app.clickup.com/t/9002019994/PRF-4895)

File: apps/evm_family/evm_priv.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * @author  Cypherock X1 Team
+ * @brief   Support for evm app internal operations
+ *          This file is defined to separate EVM's internal use functions,
+ * flows, common APIs
+ * @copyright Copyright (c) 2023 HODL TECH PTE LTD
+ * <br/> You may obtain a copy of license at <a href="https://mitcc.org/"
+ * target=_blank>https://mitcc.org/</a>
+ */
+#ifndef EVM_APP_PRIV_H
+#define EVM_APP_PRIV_H
```
Comment (ujjwal-cyph) [inline apps/evm_family/evm_priv.h:12]: make `EVM_APP_PRIV_H` to `EVM_PRIV_H`

File: apps/evm_family/evm_priv.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/**
+ * @file    evm_app_priv.h
```
Comment (ujjwal-cyph) [inline apps/evm_family/evm_priv.h:2]: should be `evm_priv.h`

File: apps/evm_family/evm_main.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
+/**
+ * @file    evm_app.h
```
Comment (ujjwal-cyph) [inline apps/evm_family/evm_main.h:2]: should be exact match with file name

File: apps/evm_family/evm_main.c [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  switch ((uint8_t)query.which_request) {
+    case EVM_QUERY_GET_PUBLIC_KEYS_TAG: {
+      delay_scr_init("Detected Ethereum query", DELAY_TIME);
+      // evm_get_public_keys(&query);
+      break;
+    }
+    default: {
+      /* In case we ever encounter invalid query, the USB event should be
+       * cleared manually */
+      usb_clear_event();
```
Comment (ujjwal-cyph) [inline apps/evm_family/evm_main.c:122]: reference the `btc_main.c` in its current state at `refactor/core-framework`

File: apps/evm_family/evm_main.h [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ * @author  Cypherock X1 Team
+ * @brief
+ * @details
+ * @copyright Copyright (c) 2023 HODL TECH PTE LTD
+ * <br/> You may obtain a copy of license at <a href="https://mitcc.org/"
+ * target=_blank>https://mitcc.org/</a>
+ *
+ */
+
+#ifndef EVM_APP_H
```
Comment (ujjwal-cyph) [inline apps/evm_family/evm_main.h:12]: `EVM_MAIN_H` instead of ~`EVM_APP_H`~

Comment (natto1784) [reply apps/evm_family/evm_main.h:12]: oh, sorry i missed it

File: apps/evm_family/evm_api.h [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+/**
+ * @brief This API encodes evm_result_t in protobuf structure.
+ * @details If the encoding is successful, then it sends the corresponding
+ * result to the host.
+ *
+ * The function ASSERTs the result of encode_evm_result internally.
+ *
+ * @param result The result which needs to be sent to the host.
+ */
+void evm_send_result(evm_result_t *result);
```
Comment (ujjwal-cyph) [inline apps/evm_family/evm_api.h:99]: let's make the function param `const`

i noticed this is also missing at other places as well. better to have it as const at all the similar places (`btc_send_result`, `manager_send_result`) in this PR itself

```suggestion
void evm_send_result(const evm_result_t *result);
```

File: apps/evm_family/evm_api.h [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+ * @brief Encodes the result with `EVM_RESULT_FIELDS` to byte-stream
+ *
+ * @param[in] result: object of populated @ref evm_result_t to be encoded
+ * @param[out] buffer: buffer to fill byte-stream into
+ * @param[in] max_buffer_len: Max length allowed for writing bytestream to
+ * buffer
+ * @param[out] bytes_written_out: bytes written to bytestream
+ * @return bool True if decoding was successful, else false
+ */
+bool encode_evm_result(evm_result_t *result,
```
Comment (ujjwal-cyph) [inline apps/evm_family/evm_api.h:57]: let's make the function param `const`

better to have it as const at all the similar places (`encode_btc_result`, `encode_manager_result`) in this PR itself

```suggestion
bool encode_evm_result(const evm_result_t *result,
```

**Review by ujjwal-cyph — APPROVED** [review]
LGTM

**Pull request merged**